soroban-env-host/src/test/invocation.rs

```diff
@@ -70,10 +71,60 @@ fn invoke_cross_contract_with_err() -> Result<(), HostError> {
     let exp_st: Status = code.into();
     assert_eq!(sv.get_payload(), exp_st.to_raw().get_payload());
 
+    let events = host.get_events()?;
+    assert_eq!(events.0.len(), 3);
+    let last_event = events.0.last();
+    match last_event {
+        Some(HostEvent::Debug(de)) => {
+            assert_eq!(de.msg, Some("contract call invocation resulted in error"));
+            assert_eq!(de.args.len(), 1);
+            if let DebugArg::Val(rv) = de.args[0] {
+                let status: Status = rv.try_into()?;
+                assert_eq!(
+                    status,
+                    Status::from_type_and_code(
+                        ScStatusType::HostObjectError,
+                        ScHostObjErrorCode::VecIndexOutOfBound as u32,
+                    )
+                );
+            } else {
+                panic!("got: {:?}, want debug arg containing Val", de);
+            }
+        }
+        _ => {
+            panic!("got: {:?}, want debug event", last_event);
+        }
+    };
```

Comment:
@graydon FYI asserting on DebugEvents is pretty painful because there's a `RawVal` embedded in there, and we can't do equality checks on `RawVal`s.